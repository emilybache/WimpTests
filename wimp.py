from enum import Enum


class Order:
    @staticmethod
    def from_data(pizza_type, size, customer, delivery):
        return Order()

    def create(self):
        return "id"


class Customer:
    @staticmethod
    def with_name(name):
        return Customer()

    @staticmethod
    def with_email_and_phone(email, phone):
        customer = Customer()
        customer.email = email
        customer.phone = phone
        return customer


class Blocklist:
    def __init__(self, emails, phones):
        self.phones = phones
        self.emails = emails

    def is_restricted(self, customer):
        if customer.email in self.emails or customer.phone in self.phones:
            return True
        return False

class Delivery(Enum):
    Collection = 1