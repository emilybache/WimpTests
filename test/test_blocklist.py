import approvaltests
import pytest

from wimp import Customer, Blocklist


@pytest.fixture
def blocklist():
    return Blocklist(emails=["email1"], phones=["phone1"])

def test_blocklisting_both_listed(blocklist):
    customer = Customer.with_email_and_phone("email1", "phone1")
    assert blocklist.is_restricted(customer)

def test_blocklisting_email_listed(blocklist):
    customer = Customer.with_email_and_phone("email1", "another_phone")
    assert blocklist.is_restricted(customer)

def test_blocklisting_phone_listed(blocklist):
    customer = Customer.with_email_and_phone("another_email", "phone1")
    assert blocklist.is_restricted(customer)

def test_blocklisting_nothing_listed(blocklist):
    customer = Customer.with_email_and_phone("another_email", "another_phone")
    assert not blocklist.is_restricted(customer)

def test_blocklisting():
    blocklist = Blocklist(emails=["email1"], phones=["phone1"])
    customers = [
        Customer.with_email_and_phone("email1", "phone1"),
        Customer.with_email_and_phone("email1", "another_phone"),
        Customer.with_email_and_phone("another_email", "phone1"),
        Customer.with_email_and_phone("another_email", "another_phone"),
    ]
    approvaltests.approvals.verify_all(f"Decide whether a customer is restricted\n according to Blocklist with contents {blocklist.emails} and {blocklist.phones}",
                                       [f"{c.email}, {c.phone} => {blocklist.is_restricted(c)}" for c in customers])
