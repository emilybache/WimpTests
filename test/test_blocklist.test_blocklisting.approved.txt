Decide whether a customer is restricted
 according to Blocklist with contents ['email1'] and ['phone1']

0) email1, phone1 => True
1) email1, another_phone => True
2) another_email, phone1 => True
3) another_email, another_phone => False
